src/cpp/session/modules/SessionHelp.cpp

```diff
@@ -732,6 +732,65 @@ r_util::RPackageInfo packageInfoForRd(const FilePath& rdFilePath)
    }
 }
 
+Error Rd2HTML(FilePath filePath, std::string* html)
+{
+   r::exec::RFunction fun(".rs.Rd2HTML", filePath.getAbsolutePath());
+
+   // add in package-specific information if available
+   r_util::RPackageInfo pkgInfo = packageInfoForRd(filePath);
+   if (!pkgInfo.empty())
+   {
+      if (!pkgInfo.name().empty())
+         fun.addParam(pkgInfo.name());
+
+      std::string macros = pkgInfo.name();
+      if (!pkgInfo.rdMacros().empty())
+         macros = macros + "," + pkgInfo.rdMacros();
+      fun.addParam(macros);
+   }
+
+   return fun.call(html);
+}
+
+template <typename Filter>
+bool handleDevRequest(const http::Request& request,
+                      const Filter& filter,
+                      http::Response* pResponse)
+{
+   std::string topic = request.queryParamValue("dev");
+   
+   r::exec::RFunction dev_topic_find("pkgload:::dev_topic_find", topic);
+   r::sexp::Protect protect;
+   SEXP res;
+   Error error = dev_topic_find.call(&res, &protect);
+   if (error || res == R_NilValue)
+   {
+      return false;
+   }
+   std::string file = CHAR(STRING_ELT(VECTOR_ELT(res, 0), 0));
```

Comment:
You might want to use 'safe' R accessor APIs, just in case this function doesn't return the value you expect. It looks like it should return a named list, e.g.

```
> pkgload:::dev_topic_find("snapshot")
$path
[1] "/Users/kevin/r/pkg/renv/man/snapshot.Rd"

$pkg
[1] "renv"
```

So `getNamedListElement()` might be useful here.

https://github.com/rstudio/rstudio/blob/3bafffe479bbc3f762a9a29d0deb99ecdf9b348e/src/cpp/r/include/r/RSexp.hpp#L221-L244